book_haven_api/views.py

```diff
@@ -24,12 +26,38 @@ def get_a_library(request, library_id):
             "detail": "The library with id " + str(library_id) + " does not exist."
         }]},status=status.HTTP_404_NOT_FOUND)
     
-@api_view(['GET'])
+@api_view(['GET', 'POST'])
 def get_library_books(request, library_id):
-    library = Library.objects.get(id=library_id)
-    books = library.book_set
-    serializer = BookSerializer(books, many=True)
-    return Response({ "data": serializer.data }, status=status.HTTP_200_OK)
+    if request.method == 'GET':
+        library = Library.objects.get(id=library_id)
+        books = library.book_set
+        serializer = BookSerializer(books, many=True)
+        return Response({ "data": serializer.data }, status=status.HTTP_200_OK)
+    
+    elif request.method == 'POST':
+        library = Library.objects.get(id=library_id)
+        api_key = config('MY_API_KEY')
+        url = 'https://www.googleapis.com/books/v1/volumes?q=isbn:' + request.data['isbn'] + '&key=' + api_key
+        response = requests.get(url)
+        if response.json()['totalItems'] == 0:
+            return Response({"errors": [{
+                "status": "422",
+                "title": "Book not created",
+                "detail": "Book could not be created because ISBN not found."
+            }]},status=status.HTTP_422_UNPROCESSABLE_ENTITY)
```

Comment:
This is an interesting one - not saying to change it, but to call out; I think typically I would expect this to be a 404, but there's an argument that since you are looking externally to see if it exists, it's not that the resource doesn't exist in your app, it just doesn't exist externally. Semantically, I'd say a 404 probably makes more sense but either one probably works since you're being explicit in the error message.